ynr/apps/api/v09/serializers.py

```diff
@@ -296,6 +296,41 @@ class Meta:
     def get_extra_fields(self, obj):
         return [{"favourite_biscuits": obj.favourite_biscuit or ""}]
 
+    def get_contact_details(self, obj):
+        ret = []
+        if obj.get_twitter_username:
+            ret.append(
+                {
+                    "contact_type": "twitter",
+                    "label": "",
+                    "note": "",
+                    "value": obj.get_twitter_username,
+                }
+            )
+        return ret
+
+    def get_identifiers(self, obj):
+        ret = []
+        if obj.get_single_identifier_of_type("theyworkforyou"):
+            ret.append(
+                {
+                    "identifier": "uk.org.publicwhip/person/{}".format(
+                        obj.get_single_identifier_of_type("theyworkforyou")
+                    ),
+                    "scheme": "uk.org.publicwhip",
+                }
+            )
+        if obj.get_single_identifier_of_type("twitter_username"):
+            ret.append(
+                {
+                    "identifier": obj.get_identifiers_of_type(
+                        "twitter_username"
+                    )[0].internal_identifier,
```

Comment:
I'm slightly confused about why one of these is using `get_single_identifier_of_type()` but the other uses `get_identifiers_of_type()[0]`. Is it because in principle we might have >1 twitter profiles attached to a person in future, but we should only ever have one TWFY ID?

Reply:
This is some bad naming that I should change – I just didn't do this while making these changes.
Basically `get_identifiers_of_type[0]` returns the `PersonIdentifier` _model_, `get_single_identifier_of_type` returns the _model value_. I expect I should move to `get_single_identifier_of_type.value` here an in other places. The initial reason for the was that I only cared about the value in the code I was writing, but this code needs the internal ID.

I'll have a think – ideas for better names welcome!